### PR TITLE
Switch from webpack to esbuild

### PR DESCRIPTION
**Pull request by mbklein**
# Summary 

Switch Meadow front end from webpack to esbuild

# Specific Changes in this PR
- Configure esbuild for `npm run deploy` and `npm run watch`
- Fix issue in `elasticsearch.js` that was preventing it from building under esbuild
- Configure Meadow to use esbuild as its file watcher

- list changes
# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.

- [x] Patch
- [ ] Minor
- [ ] Major

# Steps to Test

- Make sure the entire front end works in dev
- We won't be able to test the packaged build until it's on staging

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- Backwards-incompatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- [ ] Requires data migration
- [ ] Requires database triggers disabled during deployment/migration
- [ ] Requires reindex
- [ ] Terraform changes
  - [ ] Adds/requires new or changed Terraform variables
- [ ] Pipeline configuration changes (requires `mix meadow.pipeline.setup` run)
- [ ] Requires new variable added to `miscellany`
- [ ] Specific deployment synchronization instructions with other apps/API's
- [ ] Other specific instructions/tasks

# Tested/Verified
- [ ] End users/stakeholders

### CODE REVIEW

File: app/assets/build.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  outdir: "../priv/static/js",
+  logLevel: "info",
+  sourcemap: "linked",
+  define,
+  loader,
+  plugins,
+};
+
+async function watchMode(opts) {
+  const ctx = await esbuild.context({ ...opts, sourcemap: "inline" });
```
Comment (adamjarling) [inline app/assets/build.js:66]: This will be great if we can get sourcemaps in `dev` watch mode.

File: app/assets/js/services/elasticsearch.js [hunk 2]
```diff
@@ -7,9 +7,25 @@ export const ELASTICSEARCH_PROXY_ENDPOINT = `${host}:${port}/_search`;
 export const ELASTICSEARCH_INDEX_NAME = __ELASTICSEARCH_INDEX__;
 
 const Elasticsearch = require("elasticsearch");
+
```
Comment (adamjarling) [inline app/assets/js/services/elasticsearch.js:10]: I'd love to get `Elasticsearch` out of the client-side after we push this through, as part of the "Meadow re-evaluate" batch of client-side app work.   As long as this works and queries are firing, even if just limping along or not pretty, we'll extract it soon enough and don't need to spend too much headspace on it.

**Review by adamjarling — APPROVED** [review]
Let it rip 🚀. Nice work @mbklein

File: app/assets/js/services/elasticsearch.js (hunk 2 above)
Comment (mbklein) [reply app/assets/js/services/elasticsearch.js:10]: I agree. The newer version of the official JS client even says that [in-browser use is not supported](https://github.com/elastic/elasticsearch-js#browser).

**Pull request merged**